doc/memory.rst

```diff
@@ -60,6 +60,13 @@ A simple example:
     Running f(2)
     2
 
+.. note::
+
+  :class:`~joblib.Memory` use pickle to load saved results from the file system.
+  This means that pickle security considerations apply, see `pickle documentation
+  <https://docs.python.org/3/library/pickle.html>`_ for more details.
+  In particular, only use a cache directory if you trust it.
```

Comment:
Let's be more precise in documenting the potential attack vector:

```suggestion
  In particular, if an adversary gains write access to the cache directory,
  it could lead to arbitrary code execution by the process(es) when 
  loading tempered cached values from that folder.
```